docs: translate api useTransition (#682)
reactjs/es.react.dev: src/content/reference/react/useTransition.md

### `useTransition()` {/*usetransition*/}

> Llama a `useTransition` en el nivel superior de su componente para marcar algunas actualizaciones de estado como transiciones.

Reviewer: ```suggestion
Llama a `useTransition` en el nivel superior de tu componente para marcar algunas actualizaciones de estado como transiciones.
```

}
```

[Ver más ejemplos debajo.](#uso)

#### Parámetros {/*parameters*/}

`useTransition` no recibe ningun parámetro.

#### Devuelve {/*returns*/}

`useTransition` devuelve un _array_ con exactamente dos elementos:

1. `isPending` que indica si hay una transición pendiente.
2. [`startTransition` function](#starttransition) que permite marcar una actualización de estado como una transición.

---

> ### `startTransition` function {/*starttransition*/}

Reviewer: ```suggestion
### Función `startTransition` {/*starttransition*/}
```

}
```

#### Parámetros {/*starttransition-parameters*/}

> * `scope`: Una función que actualiza algún estado llamando a uno o varios [`set` functions.](/reference/react/useState#setstate) React inmediatamente llama`scope` sin parámetros y marca todas las actualizaciones de estado programadas sincrónicamente durante el `scope` llamada de función como transiciones. Serán [sin bloqueo](#marking-a-state-update-as-a-non-blocking-transition) y [no mostrará indicadores de carga no deseados.](#preventing-unwanted-loading-indicators)

Reviewer: ```suggestion
* `scope`: Una función que actualiza algún estado llamando a una o varias [funciones `set`.](/reference/react/useState#setstate) React inmediatamente llama a `scope` sin parámetros y marca todas las actualizaciones de estado programadas sincrónicamente durante la llamada a la función `scope` como transiciones. Serán [sin bloqueo](#marking-a-state-update-as-a-non-blocking-transition) y [no mostrarán indicadores de carga no deseados.](#preventing-unwanted-loading-indicators)
```

#### Devuelve {/*starttransition-returns*/}

`startTransition` no devuelve nada.

#### Advertencias {/*starttransition-caveats*/}

> * `useTransition` es un Gancho, por lo que sólo puede ser llamado dentro de componentes o Hooks personalizados. Si necesita iniciar una transición en otro lugar (por ejemplo, desde una biblioteca de datos), llame a la función independiente [`startTransition`](/reference/react/startTransition) en su lugar.

Reviewer: ```suggestion
* `useTransition` es un Hook, por lo que sólo puede ser llamado dentro de componentes o Hooks personalizados. Si necesitas iniciar una transición en otro lugar (por ejemplo, desde una biblioteca de datos), llama a la función independiente [`startTransition`](/reference/react/startTransition) en su lugar.
```

> * Puedes envolver una actualización en una transición sólo si tienes acceso a la función `set` de ese estado. Si desea iniciar una transición en respuesta a algunas prop o algún valor de un Hook personalizado, intente [`useDeferredValue`](/reference/react/useDeferredValue) en su lugar.

Reviewer: ```suggestion
* Puedes envolver una actualización en una transición sólo si tienes acceso a la función `set` de ese estado. Si deseas iniciar una transición en respuesta a alguna prop o algún valor de un Hook personalizado, prueba [`useDeferredValue`](/reference/react/useDeferredValue) en su lugar.
```

* La función que pases a `startTransition` debe ser síncrona. React ejecuta inmediatamente esta función, marcando como transiciones todas las actualizaciones de estado que se produzcan mientras se ejecuta. Si intentas realizar más actualizaciones de estado más tarde (por ejemplo, en un tiempo de espera), no se marcarán como transiciones.

> * Una actualización de estado marcada como transición será interrumpida por otras actualizaciones de estado. Por ejemplo, si actualizas un componente gráfico dentro de una transición, pero luego empiezas a escribir en un input mientras el gráfico está en medio de un re-renderizado, React reiniciará el trabajo de renderizado en el componente gráfico después de gestionar la actualización del input.

Reviewer: ```suggestion
* Una actualización de estado marcada como transición será interrumpida por otras actualizaciones de estado. Por ejemplo, si actualizas un componente gráfico dentro de una transición, pero luego empiezas a escribir en un input mientras el gráfico está en medio de un rerenderizado, React reiniciará el trabajo de renderizado en el componente gráfico después de gestionar la actualización del input.
```

* Las actualizaciones de transición no pueden utilizarse para controlar las entradas de texto.

* Si hay varias transiciones en curso, React las agrupa. Se trata de una limitación que probablemente se eliminará en una versión futura.

---

## Uso {/*usage*/}

### Marcar una actualización de estado como transición no bloqueante {/*marking-a-state-update-as-a-non-blocking-transition*/}

> Llama a `useTransition` en el nivel superior de tu componente para marcar las actualizaciones de estado como no bloqueantes. *transitions*.

Reviewer: ```suggestion
Llama a `useTransition` en el nivel superior de tu componente para marcar las actualizaciones de estado como *transiciones* no bloqueantes.
```

}
```

`useTransition` devuelve un array con exactamente dos elementos:

> 1. <CodeStep step={1}>`isPending` flag</CodeStep> que le indica si hay una transición pendiente.

Reviewer: ```suggestion
1. <CodeStep step={1}>`isPending` flag</CodeStep> que te indica si hay una transición pendiente.
```

2. <CodeStep step={2}>`startTransition` function</CodeStep> que permite marcar una actualización de estado como una transición.

> A continuación, puede marcar una actualización de estado como una transición de esta manera:

Reviewer: ```suggestion
A continuación, puedes marcar una actualización de estado como una transición de esta manera:
```

}
```

Las transiciones permiten mantener la capacidad de respuesta de las actualizaciones de la interfaz de usuario incluso en dispositivos lentos.

Con una transición, la interfaz de usuario mantiene su capacidad de respuesta en medio de una nueva renderización. Por ejemplo, si el usuario hace clic en una pestaña pero luego cambia de opinión y hace clic en otra, puede hacerlo sin esperar a que termine la primera renderización.

> <Recipes titleText="Diferencia entre useTransition y las actualizaciones de estado normales" titleId="ejemplos">

Reviewer: ```suggestion
<Recipes titleText="Diferencia entre useTransition y las actualizaciones de estado normales" titleId="examples">
```

---

### Visualización de un estado visual pendiente durante la transición {/*displaying-a-pending-visual-state-during-the-transition*/}

> Puede utilizar el valor booleano `isPending` devuelto por `useTransition` para indicar al usuario que una transición está en curso. Por ejemplo, el botón de pestaña puede tener un estado visual especial "pendiente":

Reviewer: ```suggestion
Puedes utilizar el valor booleano `isPending` devuelto por `useTransition` para indicar al usuario que una transición está en curso. Por ejemplo, el botón de la pestaña puede tener un estado visual especial "pendiente":
```

---

### Evitar indicadores de carga no deseados {/*preventing-unwanted-loading-indicators*/}

> En este ejemplo, el componente `PostsTab` obtiene algunos datos utilizando una fuente de datos [Suspense-enabled](/reference/react/Suspense). Al hacer clic en la pestaña _"Posts"_, el componente `PostsTab` se *suspende*, haciendo que aparezca el fallback de carga más cercano:

Reviewer: ```suggestion
En este ejemplo, el componente `PostsTab` obtiene algunos datos utilizando una fuente de datos [preparada par usarse con Suspense](/reference/react/Suspense). Al hacer clic en la pestaña _"Posts"_, el componente `PostsTab` se *suspende*, haciendo que aparezca el fallback de carga más cercano:
```

</Sandpack>

[Más información sobre el uso de transiciones con Suspense.](/reference/react/Suspense#preventing-already-revealed-content-from-hiding)

<Note>

> Las transiciones sólo "esperarán" el tiempo suficiente para evitar ocultar el contenido *ya revelado* (como el contenedor de la pestaña). Si la pestaña Entradas tuviera un [`<Suspense>` boundary anidado,](/reference/react/Suspense#revealing-nested-content-as-it-loads) la transición no la "esperaría".

Reviewer: https://github.com/reactjs/es.react.dev/wiki/Glosario
```suggestion
Las transiciones sólo "esperarán" el tiempo suficiente para evitar ocultar el contenido *ya revelado* (como el contenedor de la pestaña). Si la pestaña Entradas tuviera una [barrera de `<Suspense>` anidada,](/reference/react/Suspense#revealing-nested-content-as-it-loads) la transición no la "esperaría".
```

</Note>

---

> ### Construir un router con Suspense {/*building-a-suspense-enabled-router*/}

Reviewer: ```suggestion
### Construir un enrutador preparado para Suspense {/*building-a-suspense-enabled-router*/}
```

> Si estás construyendo un framework React o un router, te recomendamos marcar las navegaciones de página como transiciones.

Reviewer: ```suggestion
Si estás construyendo un framework de React o un enrutador, te recomendamos marcar las navegaciones de página como transiciones.
```

<Note>

> [Suspense habilitado](/reference/react/Suspense) se espera que los routers envuelvan las actualizaciones de navegación en transiciones por defecto.

Reviewer: ```suggestion
Se espera que los enredadores [preparados para Suspense](/reference/react/Suspense) envuelvan las actualizaciones de navegación en transiciones por defecto.
```

return <input value={text} onChange={handleChange} />;
```

> Esto se debe a que las transiciones no se bloquean, pero la actualización de una entrada en respuesta al evento de cambio debe producirse de forma sincrónica. Si desea ejecutar una transición en respuesta a la escritura, tiene dos opciones:

Reviewer: ```suggestion
Esto se debe a que las transiciones son no bloqueantes, pero la actualización de una entrada en respuesta al evento de cambio debe producirse de forma sincrónica. Si deseas ejecutar una transición en respuesta a la escritura, tiene dos opciones:
```

1. Puedes declarar dos variables de estado separadas: una para el estado de la entrada (que siempre se actualiza de forma sincrónica), y otra que actualizarás en una transición. Esto te permite controlar la entrada utilizando el estado síncrono, y pasar la variable de estado de transición (que "irá por detrás" de la entrada) al resto de tu lógica de renderizado.

> 2. Alternativamente, puedes tener una variable de estado, y añadir [`useDeferredValue`](/reference/react/useDeferredValue) que "irá por detrás" del valor real. Se activarán re-renders no bloqueantes para "ponerse al día" con el nuevo valor de forma automática.

Reviewer: ```suggestion
2. Alternativamente, puedes tener una variable de estado, y añadir [`useDeferredValue`](/reference/react/useDeferredValue) que "irá por detrás" del valor real. Se activarán rerenderizados no bloqueantes para "ponerse al día" con el nuevo valor de forma automática.
```

---

### React no trata mi actualización de estado como una transición {/*react-doesnt-treat-my-state-update-as-a-transition*/}

> Cuando envuelva una actualización de estado en una transición, asegúrese de que ocurre *durante* la llamada `startTransition`:

Reviewer: https://github.com/reactjs/es.react.dev/wiki/Gu%C3%ADa-de-estilo#usted-tú-y-vos
```suggestion
Cuando envuelvas una actualización de estado en una transición, asegúrate de que ocurre *durante* la llamada `startTransition`:
```

console.log(3);
```

> **Se espera que imprima 1, 2, 3.** La función que pasas a `startTransition` no se retrasa. Al contrario que con el navegador `setTimeout`, no ejecuta el callback más tarde. React ejecuta tu función inmediatamente, pero cualquier actualización de estado programada *mientras se está ejecutando* se marca como transición. Puedes imaginar que funciona así:

Reviewer: ```suggestion
**Se espera que imprima 1, 2, 3.** La función que pasas a `startTransition` no se retrasa. Al contrario que con el `setTimeout` del navegador, no ejecuta el callback más tarde. React ejecuta tu función inmediatamente, pero cualquier actualización de estado programada *mientras se está ejecutando* se marca como transición. Puedes imaginar que funciona así:
```
